diff --git a/WORKSPACE b/WORKSPACE
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -355,6 +355,12 @@ new_local_repository(
     path = "C:\\opencv\\build",
 )
 
+new_local_repository(
+    name = "godot-cpp",
+    build_file = "@//mediapipe/GDMP:godot-cpp.BUILD",
+    path = "../godot-cpp",
+)
+
 http_archive(
     name = "android_opencv",
     build_file = "@//third_party:opencv_android.BUILD",
diff --git a/mediapipe/util/BUILD b/mediapipe/util/BUILD
--- a/mediapipe/util/BUILD
+++ b/mediapipe/util/BUILD
@@ -190,6 +190,7 @@ cc_library(
         "resource_util_internal.h",
     ] + select({
         "//conditions:default": ["resource_util_default.cc"],
+        "//mediapipe/GDMP:godot": [],
         "//mediapipe:android": ["resource_util_android.cc"],
         "//mediapipe/framework:android_no_jni": ["resource_util_loonix.cc"],
         "//mediapipe:ios": ["resource_util_apple.cc"],
@@ -223,6 +224,9 @@ cc_library(
         "//conditions:default": [
             "@com_google_absl//absl/flags:flag",
         ],
+        "//mediapipe/GDMP:godot": [
+            "//mediapipe/GDMP/util:resource_util_godot",
+        ],
         "//mediapipe:android": [
             "//mediapipe/util/android:asset_manager_util",
             "//mediapipe/util/android/file/base",
