diff --git a/WORKSPACE b/WORKSPACE
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -2,6 +2,17 @@ workspace(name = "mediapipe")
 
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
+local_repository(
+    name = "GDMP",
+    path = "../",
+)
+
+new_local_repository(
+    name = "godot-cpp",
+    build_file = "@GDMP//GDMP:godot-cpp.BUILD",
+    path = "../godot-cpp",
+)
+
 # Protobuf expects an //external:python_headers target
 bind(
     name = "python_headers",
