diff --git a/WORKSPACE b/WORKSPACE
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -2,6 +2,47 @@ workspace(name = "mediapipe")
 
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
+local_repository(
+    name = "GDMP",
+    path = "../",
+)
+
+new_local_repository(
+    name = "godot-cpp",
+    build_file = "@GDMP//GDMP:godot-cpp.BUILD",
+    path = "../godot-cpp",
+)
+
+http_archive(
+    name = "emsdk",
+    strip_prefix = "emsdk-3.1.64/bazel",
+    url = "https://github.com/emscripten-core/emsdk/archive/3.1.64.tar.gz",
+)
+
+load("@emsdk//:deps.bzl", emsdk_deps = "deps")
+emsdk_deps()
+
+load("@emsdk//:emscripten_deps.bzl", emsdk_emscripten_deps = "emscripten_deps")
+emsdk_emscripten_deps(emscripten_version = "3.1.64")
+
+load("@emsdk//:toolchains.bzl", "register_emscripten_toolchains")
+register_emscripten_toolchains(
+    cache = {
+        "configuration": ["--pic"],
+        "targets": [
+            "crtbegin",
+        ],
+    },
+)
+
+http_archive(
+    name = "hedron_compile_commands",
+    url = "https://github.com/hedronvision/bazel-compile-commands-extractor/archive/0e990032f3c5a866e72615cf67e5ce22186dcb97.tar.gz",
+    strip_prefix = "bazel-compile-commands-extractor-0e990032f3c5a866e72615cf67e5ce22186dcb97",
+)
+load("@hedron_compile_commands//:workspace_setup.bzl", "hedron_compile_commands_setup")
+hedron_compile_commands_setup()
+
 # Protobuf expects an //external:python_headers target
 bind(
     name = "python_headers",
