diff --git a/MODULE.bazel b/MODULE.bazel
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -7,6 +7,8 @@ single_version_override(
     version = "0.0.11",
 )
 
+bazel_dep(name = "apple_support", version = "2.6.1", repo_name = "build_bazel_apple_support")
+
 bazel_dep(name = "rules_cc", version = "0.2.9")
 single_version_override(
     module_name = "rules_cc",
@@ -27,5 +29,4 @@ bazel_dep(name = "aspect_rules_js", version = "2.7.0")
 bazel_dep(name = "aspect_bazel_lib", version = "2.21.2")
 bazel_dep(name = "aspect_rules_rollup", version = "2.0.0")
 bazel_dep(name = "aspect_rules_ts", version = "3.7.0")
-bazel_dep(name = "apple_support", version = "2.6.1", repo_name = "build_bazel_apple_support")
 bazel_dep(name = "rules_apple", version = "3.16.1", repo_name = "build_bazel_rules_apple")
