diff --git a/WORKSPACE b/WORKSPACE
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -81,11 +81,12 @@ http_archive(
     ],
     patches = [
         "@//third_party:com_github_glog_glog.diff",
+        "@//third_party:com_github_glog_glog_windows_patch.diff",
     ],
-    sha256 = "8a83bf982f37bb70825df71a9709fa90ea9f4447fb3c099e1d720a439d88bad6",
-    strip_prefix = "glog-0.6.0",
+    sha256 = "170d08f80210b82d95563f4723a15095eff1aad1863000e8eeb569c96a98fefb",
+    strip_prefix = "glog-3a0d4d22c5ae0b9a2216988411cfa6bf860cc372",
     urls = [
-        "https://github.com/google/glog/archive/v0.6.0.tar.gz",
+        "https://github.com/google/glog/archive/3a0d4d22c5ae0b9a2216988411cfa6bf860cc372.zip",
     ],
 )
 
diff --git a/third_party/BUILD b/third_party/BUILD
--- a/third_party/BUILD
+++ b/third_party/BUILD
@@ -40,29 +40,8 @@ cc_library(
     name = "glog",
     visibility = ["//visibility:public"],
     deps = select({
-        "//mediapipe:android_x86": [
-            "@com_github_glog_glog_no_gflags//:glog",
-        ],
-        "//mediapipe:android_x86_64": [
-            "@com_github_glog_glog_no_gflags//:glog",
-        ],
-        "//mediapipe:android_arm": [
-            "@com_github_glog_glog_no_gflags//:glog",
-        ],
-        "//mediapipe:android_arm64": [
-            "@com_github_glog_glog_no_gflags//:glog",
-        ],
-        "//mediapipe:ios": [
-            "@com_github_glog_glog_no_gflags//:glog",
-        ],
-        "//mediapipe:macos": [
-            "@com_github_glog_glog//:glog",
-        ],
-        "//mediapipe:windows": [
-            "@com_github_glog_glog_windows//:glog",
-        ],
         "//conditions:default": [
-            "@com_github_glog_glog//:glog",
+            "@com_github_glog_glog_no_gflags//:glog",
         ],
     }),
 )
