diff --git a/WORKSPACE b/WORKSPACE
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -765,33 +806,29 @@ new_local_repository(
 http_archive(
     name = "linux_halide",
     build_file = "@//third_party:halide.BUILD",
-    sha256 = "d290fadf3f358c94aacf43c883de6468bb98883e26116920afd491ec0e440cd2",
-    strip_prefix = "Halide-15.0.1-x86-64-linux",
-    urls = ["https://github.com/halide/Halide/releases/download/v15.0.1/Halide-15.0.1-x86-64-linux-4c63f1befa1063184c5982b11b6a2cc17d4e5815.tar.gz"],
+    strip_prefix = "Halide-18.0.0-x86-64-linux",
+    urls = ["https://github.com/halide/Halide/releases/download/v18.0.0/Halide-18.0.0-x86-64-linux-8c651b459a4e3744b413c23a29b5c5d968702bb7.tar.gz"],
 )
 
 http_archive(
     name = "macos_x86_64_halide",
     build_file = "@//third_party:halide.BUILD",
-    sha256 = "48ff073ac1aee5c4aca941a4f043cac64b38ba236cdca12567e09d803594a61c",
-    strip_prefix = "Halide-15.0.1-x86-64-osx",
-    urls = ["https://github.com/halide/Halide/releases/download/v15.0.1/Halide-15.0.1-x86-64-osx-4c63f1befa1063184c5982b11b6a2cc17d4e5815.tar.gz"],
+    strip_prefix = "Halide-18.0.0-x86-64-osx",
+    urls = ["https://github.com/halide/Halide/releases/download/v18.0.0/Halide-18.0.0-x86-64-osx-8c651b459a4e3744b413c23a29b5c5d968702bb7.tar.gz"],
 )
 
 http_archive(
     name = "macos_arm_64_halide",
     build_file = "@//third_party:halide.BUILD",
-    sha256 = "db5d20d75fa7463490fcbc79c89f0abec9c23991f787c8e3e831fff411d5395c",
-    strip_prefix = "Halide-15.0.1-arm-64-osx",
-    urls = ["https://github.com/halide/Halide/releases/download/v15.0.1/Halide-15.0.1-arm-64-osx-4c63f1befa1063184c5982b11b6a2cc17d4e5815.tar.gz"],
+    strip_prefix = "Halide-18.0.0-arm-64-osx",
+    urls = ["https://github.com/halide/Halide/releases/download/v18.0.0/Halide-18.0.0-arm-64-osx-8c651b459a4e3744b413c23a29b5c5d968702bb7.tar.gz"],
 )
 
 http_archive(
     name = "windows_halide",
     build_file = "@//third_party:halide.BUILD",
-    sha256 = "61fd049bd75ee918ac6c30d0693aac6048f63f8d1fc4db31001573e58eae8dae",
-    strip_prefix = "Halide-15.0.1-x86-64-windows",
-    urls = ["https://github.com/halide/Halide/releases/download/v15.0.1/Halide-15.0.1-x86-64-windows-4c63f1befa1063184c5982b11b6a2cc17d4e5815.zip"],
+    strip_prefix = "Halide-18.0.0-x86-64-windows",
+    urls = ["https://github.com/halide/Halide/releases/download/v18.0.0/Halide-18.0.0-x86-64-windows-8c651b459a4e3744b413c23a29b5c5d968702bb7.zip"],
 )
 
 http_archive(
diff --git a/third_party/halide/halide.bzl b/third_party/halide/halide.bzl
--- a/third_party/halide/halide.bzl
+++ b/third_party/halide/halide.bzl
@@ -97,6 +97,8 @@ _HALIDE_TARGET_CONFIG_SETTINGS_MAP = {
     "x86-64-windows": ["@mediapipe//mediapipe:windows"],
     # Linux
     "x86-64-linux": ["@mediapipe//mediapipe:linux"],
+    # WebAssembly
+    "wasm-32-wasmrt": ["@mediapipe//mediapipe:emscripten"],
     # Deliberately no //condition:default clause here.
 }
 
@@ -293,6 +295,8 @@ def _gengen_impl(ctx):
     # Append the features common to everything.
     target_features.append("c_plus_plus_name_mangling")
     target_features.append("no_runtime")
+    if base_target == "wasm-32-wasmrt":
+        target_features.append("wasm_threads")
 
     # Make it all neat and tidy.
     target_features = sorted(collections.uniq(target_features))
