diff --git a/MODULE.bazel b/MODULE.bazel
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -49,8 +49,8 @@ bazel_dep(name = "re2", version = "2025-11-05", repo_name = "com_googlesource_co
 bazel_dep(name = "protobuf", repo_name = "com_google_protobuf")
 archive_override(
     module_name = "protobuf",
-    strip_prefix = "protobuf-6.31.1",
-    urls = ["https://github.com/protocolbuffers/protobuf/archive/refs/tags/v6.31.1.zip"],
+    strip_prefix = "protobuf-32.0",
+    urls = ["https://github.com/protocolbuffers/protobuf/archive/refs/tags/v32.0.zip"],
 )
 
 python = use_extension("@rules_python_bzlmod//python/extensions:python.bzl", "python")
