diff --git a/mediapipe/calculators/tensor/BUILD b/mediapipe/calculators/tensor/BUILD
--- a/mediapipe/calculators/tensor/BUILD
+++ b/mediapipe/calculators/tensor/BUILD
@@ -1042,7 +1042,6 @@ cc_library(
         "@litert//tflite/c:common",
         "@litert//tflite/core:framework",
     ] + select({
-        "//mediapipe:emscripten": [],
         "//conditions:default": [
             "//mediapipe/util:cpu_util",
         ],
@@ -1120,6 +1119,7 @@ cc_library(
     name = "inference_calculator_gl_if_compute_shader_available",
     deps = selects.with_or({
         ":compute_shader_unavailable": [],
+        "//mediapipe:emscripten": [],
         "//conditions:default": [
             ":inference_calculator_gl",
             ":inference_calculator_gl_advanced",
