diff --git a/mediapipe/framework/BUILD b/mediapipe/framework/BUILD
--- a/mediapipe/framework/BUILD
+++ b/mediapipe/framework/BUILD
@@ -940,6 +940,12 @@ cc_library(
     name = "output_stream_handler",
     srcs = ["output_stream_handler.cc"],
     hdrs = ["output_stream_handler.h"],
+    copts = select({
+        "//conditions:default": [],
+        "//mediapipe:windows": [
+            "/std:c++20",
+        ],
+    }),
     visibility = [
         ":mediapipe_internal",
     ],
diff --git a/mediapipe/util/tflite/BUILD b/mediapipe/util/tflite/BUILD
--- a/mediapipe/util/tflite/BUILD
+++ b/mediapipe/util/tflite/BUILD
@@ -152,6 +152,12 @@ cc_library_with_tflite(
     tflite_deps = [
         "@org_tensorflow//tensorflow/lite:framework_stable",
     ],
+    copts = select({
+        "//conditions:default": [],
+        "//mediapipe:windows": [
+            "/std:c++20",
+        ],
+    }),
     visibility = ["//visibility:public"],
     deps = [
         "//mediapipe/framework:resources",
diff --git a/mediapipe/tasks/cc/text/language_detector/BUILD b/mediapipe/tasks/cc/text/language_detector/BUILD
--- a/mediapipe/tasks/cc/text/language_detector/BUILD
+++ b/mediapipe/tasks/cc/text/language_detector/BUILD
@@ -20,6 +20,12 @@ cc_library(
     name = "language_detector",
     srcs = ["language_detector.cc"],
     hdrs = ["language_detector.h"],
+    copts = select({
+        "//conditions:default": [],
+        "//mediapipe:windows": [
+            "/std:c++20",
+        ],
+    }),
     visibility = ["//visibility:public"],
     deps = [
         "//mediapipe/framework/api2:builder",
